### PR TITLE
refactor: use prost instead of protobuf and protoc-rust

### PR DESCRIPTION
**Pull request by guidota**
## Description

Introduces `prost` crate and replaces `protobuf + protoc-rust`. It uses `bytes` crate for serialization which has a better API and is less error-prone (for instance, serializing cannot fail).
This change enables other future implementations for code generation.

Other changes:
- replaces `make` with `just` for project-specific commands.
- moves integration example to its own crate (allowing a specific build script)

### CODE REVIEW

File: examples/integration/build.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+extern crate prost_build;
+use std::io::Result;
+
+fn main() -> Result<()> {
```
Comment (lauti7) [inline examples/integration/build.rs:4]: cool!

File: examples/integration/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+pub mod codegen;
+pub mod service;
+include!(concat!(env!("OUT_DIR"), "/_.rs"));
```
Comment (lauti7) [inline examples/integration/src/lib.rs:3]: what does this do?

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
 /target
+/examples/integration/target
```
Comment (moliva) [inline .gitignore:2]: Maybe `**/target` for a more extensible approach?

**Review by moliva — APPROVED** [review]
Yass!

**Pull request merged**